src/fsharp/vs/service.fs

```diff
@@ -2101,7 +2113,13 @@ module Helpers =
     let AreSubsumable3((fileName1:string,_,o1:FSharpProjectOptions),(fileName2:string,_,o2:FSharpProjectOptions)) =
         (fileName1 = fileName2)
         && FSharpProjectOptions.AreSubsumable(o1,o2)
-        
+
+type FileName = string
+type Source = string        
+type FilePath = string
+type ProjectPath = string
+type FileVersion = int
```

Comment:
OK - I'd be very much in favour of systematically applying type safety (not aliases) to all of the above in the FCS and indeed the entire F# compiler codebase.  We should always have done that.